module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -482,9 +482,25 @@ public function getHolding($bibId, array $patron = null, array $options = [])
             $itemResponse = $this->makeRequest('GET', '/item-storage/items', $query);
             $itemBody = json_decode($itemResponse->getBody());
             $notesFormatter = function ($note) {
-                return $note->note ?? '';
+                return !$note->staffOnly && $note->note ? $note->note : '';
             };
+            $textFormatter = function ($supplement) {
+                $format = '%s %s';
+                $supStat = $supplement->statement;
+                $supNote = $supplement->note;
+                $statement = sprintf($format, $supStat, $supNote);
+                return $statement ?? '';
+            };
+            $holdingNotes = array_map($notesFormatter, $holding->notes ?? []);
+            $hasHoldingNotes = !empty(implode($holdingNotes));
+            $holdingsStatements = $holding->holdingsStatements;
+            $holdingsSupplements = $holding->holdingsStatementsForSupplements;
+            $holdingsIndexes = $holding->holdingsStatementsForIndexes;
+            $issues = array_map($textFormatter, $holdingsStatements ?? []);
+            $supplements = array_map($textFormatter, $holdingsSupplements ?? []);
+            $indexes = array_map($textFormatter, $holdingsIndexes ?? []);
             foreach ($itemBody->items as $item) {
+                $itemNotes = array_map($notesFormatter, $item->notes ?? []);
```

Comment:
Is it worth wrapping an `array_filter` around this to strip out empty entries resulting from staff only notes?

Reply:
I don't see why not.